src/library.js

```diff
@@ -1205,14 +1206,17 @@ LibraryManager.library = {
   // an invalid index into the FUNCTION_TABLE, so something has gone wrong.
   __cxa_end_catch__deps: ['__cxa_free_exception', '$EXCEPTIONS'],
   __cxa_end_catch: function() {
-    if (___cxa_end_catch.rethrown) {
-      ___cxa_end_catch.rethrown = false;
+    var ptr = EXCEPTIONS.caught.pop();
+    if (ptr && EXCEPTIONS.infos[EXCEPTIONS.deAdjust(ptr)].rethrown) {
```

Comment:
question: why not continue in the code as before, but set `rethrown` to false (if the info exists)? We decref later down anyhow. i guess i'm not sure why things diverge here, when i'm hoping all we need is to mark a rethrown exception as no longer rethrown?